architecture/image-controller.md

```diff
@@ -1,54 +1,152 @@
 
-To create an image repository for a Component, apply this YAML code:
+### To create an general purpose image repository, apply this YAML code:
+```yaml
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ImageRepository
+metadata:
+    name: imagerepository-for-component-sample
+    namespace: test-ns
+```
+As a result, a public image repository `quay.io/my-org/test-ns/imagerepository-sample` will be created,
+based on `$DEFAULT_REGISTRY_ORG/$USER_NAMESPACE/$IMAGE_REPOSITORY_NAME`.
+
+Two robot accounts and corresponding Kubernetes Secrets for push and pull are created.
 
+### To create an image repository for a Component, apply this YAML code:
 ```yaml
 apiVersion: appstudio.redhat.com/v1alpha1
 kind: ImageRepository
 metadata:
     name: imagerepository-for-component-sample
     namespace: test-ns
+    annotations:
+        image-controller.appstudio.redhat.com/update-component-image: 'true'
     labels:
         appstudio.redhat.com/component: my-component
         appstudio.redhat.com/application: my-app
 ```
+As a result, a public image repository `quay.io/my-org/test-ns/my-component` will be created,
+based on `$DEFAULT_REGISTRY_ORG/$USER_NAMESPACE/$COMPONENT_NAME`.
+
+Two robot accounts and corresponding Kubernetes Secrets for push and pull are created.
+
+Annotation `image-controller.appstudio.redhat.com/update-component-image` is required when using
+ImageRepository with Component, as it will set Component's `spec.containerImage` allowing
+Build service controller to continue.
+
+### User defined repository name
+One may request custom image repository name by setting `spec.image.name` field upon
+the ImageRepository object creation, but it will always be prepended by
+`$DEFAULT_REGISTRY_ORG/$USER_NAMESPACE`.
 
-By default, a public image repository is created, and two robot accounts and corresponding Kubernetes Secret objects are created for pull and push individually. All these artifacts information are recorded in the `.status` field.
+e.g. when `spec.image.name` is set to `my-repository` final repository url will be
+`$DEFAULT_REGISTRY_ORG/$USER_NAMESPACE/my-repository`.
 
+Note, it's not possible to change image repository name after creation.
+Any changes to the field will be reverted by the operator.
+
+### Setting quay.io notifications
+Notifications can be set with:
+```yaml
+spec:
+  notifications:
+  - config:
+      url: https://bombino.api.redhat.com/v1/sbom/quay/push
+    event: repo_push
+    method: webhook
+    title: SBOM-event-to-Bombino
+```
+
+### Changing repository visibility
+By default, a public image repository is created.
 To change the image repository visibility, set `public` or `private` to `.spec.image.visibility`.
 
-To regenerate pull and push token, set `true` to `.spec.credentials.regenerate-token`
+### Credentials rotation for repository
+To regenerate tokens push (also pull when used with Component CR), set `true` to `.spec.credentials.regenerate-token`, it will also re-create secrets.
```

Comment:
```suggestion
To regenerate push tokens (also pull when used with Component CR), set `true` to `.spec.credentials.regenerate-token`, it will also re-create secrets.
```
I also don't quite understand why pull tokens are tied to the Component CR. Does image-controller not generate pull tokens if the ImageRepository is not connected with a Component?

Reply:
oh I thought I've changed it everywhere

will fix